src/retry/client.rs

```diff
@@ -208,6 +223,42 @@ where
         })
     }
 
+    pub fn dispatch_put<T>(self) -> Result<Option<T>>
+    where
+        T: for<'de> serde::Deserialize<'de>,
+    {
+        let url = reqwest::Url::parse(self.url.as_str()).chain_err(|| "failed to parse uri")?;
+
+        self.retry.clone().retry(|attempt| {
+            let mut builder = blocking::Client::new()
+                .put(url.clone())
+                .headers(self.headers.clone())
+                .header(header::CONTENT_TYPE, self.d.content_type());
+            if let Some(ref content) = self.body {
+                builder = builder.body(content.clone());
+            };
+            let req = builder
+                .build()
+                .chain_err(|| "failed to build PUT request")?;
+
+            info!("Putting {}: Attempt #{}", req.url(), attempt + 1);
+            let response = self
+                .client
+                .execute(req)
+                .chain_err(|| "failed to PUT request")?;
+            let status = response.status();
+            if status.is_success() {
+                info!("Put successful");
```

Comment:
Leftover from debug?

If you want to leave this in place but not visible by default, just make this a `trace!()`. Also, it's uppercase `PUT`.